src/extension/ddappsec.c

```diff
@@ -208,8 +209,17 @@ static PHP_MSHUTDOWN_FUNCTION(ddappsec)
     return SUCCESS;
 }
 
+static void dd_rinit_once(void) {
+    dd_phpobj_load_env_values();
+}
+
+static pthread_once_t dd_rinit_once_control = PTHREAD_ONCE_INIT;
+
 static PHP_RINIT_FUNCTION(ddappsec)
 {
+    // Things that should only run on the first RINIT
+    pthread_once(&dd_rinit_once_control, dd_rinit_once);
```

Comment:
Should this be done the first thing on the RINIT? It makes sense we check globals here before checking if the extension is enabled on the next line